presentation/src/main/java/com/architect/coders/mu8/comics/ComicsViewModel.kt

```diff
@@ -2,43 +2,41 @@ package com.architect.coders.mu8.comics
 
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import com.architect.coders.mu8.comics.ComicsViewModel.UiModel.*
+import com.architect.coders.mu8.utils.Event
 import com.architect.coders.mu8.utils.ScopedViewModel
 import com.architect.codes.mu8.comics.ComicUseCase
 import com.architect.codes.mu8.comics.Comic
+import com.architect.codes.mu8.utils.ERROR_INTERNET_MESSAGE
 import kotlinx.coroutines.launch
 import java.io.IOException
 
 class ComicsViewModel(private val useCase: ComicUseCase) : ScopedViewModel() {
 
-    sealed class UiModel {
-        object ShowLoading : UiModel()
-        object InternetError : UiModel()
-        class LoadData(val comics: List<Comic>) : UiModel()
-        class NavigateTo(val comic: Comic) : UiModel()
-    }
-
-    private val _model = MutableLiveData<UiModel>()
-
-    val model: LiveData<UiModel>
-        get() {
-            if (_model.value == null) loadData()
-            return _model
-        }
-
-    private fun loadData() {
+    init {
         launch {
-            _model.value = ShowLoading
-
+            _loading.value = true
             try {
-                _model.value = LoadData(useCase())
+                _comics.value = useCase()
             } catch (e: IOException) {
-                _model.value = InternetError
+                _messageError.value = ERROR_INTERNET_MESSAGE
             }
+            _loading.value = false
         }
     }
 
+    private val _comics = MutableLiveData<List<Comic>>()
+    val comics: LiveData<List<Comic>> get() = _comics
+
+    private val _loading = MutableLiveData<Boolean>()
+    val loading: LiveData<Boolean> get() = _loading
+
+    private val _navegateTo = MutableLiveData<Event<Comic>>()
```

Comment:
Please  check, that Events, is an Observable, then we  dont need the external MutableLiveData